packages/cloud-config/oem/08_boot_assessment.yaml

```diff
@@ -0,0 +1,112 @@
+# The boot assessment works in this way:
+# - After we upgrade, we set a grub env sentinel variable indicating that we did run an upgrade
+# - At the first boot, if we did an upgrade, we set another sentinel variable, which indicates we are trying to boot
+# - If we boot fine, we remove both sentinels
+# - If we get back again at the GRUB menu, a failure must have occurred and we select the fallback entry, creating also
+#   sentinels files and a specific cmdline option indicating we failed booting after an upgrade
+
+name: "Boot assessment"
+stages:
+    boot.before:
+    # Remove any grub sentinel and add
+    # System sentinel to /run to indicate if we are booting after a failed upgrade.
+    # This needs to run only on the active partition.
+    - name: "Remove upgrade sentinels"
+      if: |
+          cat /proc/cmdline | grep -q "active.img"
+      commands:
+      - |
+          sudo mount -o rw,remount /run/initramfs/cos-state
+          grub2-editenv /run/initramfs/cos-state/boot_assessment set osupgrade=
+          grub2-editenv /run/initramfs/cos-state/boot_assessment set tentative=
+          sudo mount -o ro,remount /run/initramfs/cos-state
+    - name: "Create upgrade failure sentinel if necessary"
+      if: |
+          cat /proc/cmdline | grep -q "upgrade_failure"
+      files:
+      - path: /run/cos/upgrade_failure
```

Comment:
This adds a new sentinel file, `/run/cos/upgrade_failure` when we are booting from fallback after a failed upgrade. Needs some rooms in our docs